doc/src/manual/modules.md

```diff
@@ -108,7 +108,11 @@ To mark a name as public without exporting it into the namespace of folks who ca
 one can use `public` instead of `export`. This marks the public name(s) as part of the public API,
 but does not have any namespace implications. The `public` keyword is only available in Julia 1.11
 and above. To maintain compatibility with Julia 1.10 and below, use the `@compat` macro from the
-[Compat](https://github.com/JuliaLang/Compat.jl) package.
+[Compat](https://github.com/JuliaLang/Compat.jl) package, or the version-aware construct
+
+```julia
+VERSION >= v"1.11.0-DEV.469" && eval(Meta.parse("public a, b, c"))
```

Comment:
How about
```julia
@static if VERSION >= v"1.11.0-DEV.469"
    @eval public a, b, c
end
```
?  Bit more verbose, but doesn't need to go through `Meta.parse`.

Reply:
`public` is a conditional keyword and parses as an identifier in this case, so this will throw

```julia-repl
julia> @static if VERSION >= v"1.11.0-DEV.469"
           @eval public a, b, c
       end
ERROR: UndefVarError: `public` not defined in `Main`
Suggestion: check for spelling errors or missing imports.
Stacktrace:
 [1] top-level scope
   @ REPL[3]:2
```